From 0e4e6211f011b20a6c8d665972893bab49950141 Mon Sep 17 00:00:00 2001
From: Daniel Goehring <dgoehrin@os.amperecomputing.com>
Date: Thu, 16 Apr 2020 15:43:37 -0400
Subject: [PATCH 14/32] QS|MQ: Emulation user config files [DO NOT UPSTREAM]

Add user configuration files for Ampere Altra ("Quicksilver")
and Ampere Altra Max ("Mystique") in Emulation.

Protium configuration uses an Olimex ARM-USB-OCD_H JTAG probe.
https://www.olimex.com/Products/ARM/JTAG/ARM-USB-OCD-H/

OpenOCD configured for the following QS options
1. JTAG DPI Emulation interface (Palladium)
2. JTAG Olimex (Protium)
3. One and two socket platforms

Tested on the following system platforms
* Palladium 1S2C1M and 2S2C1M
* Protium

Change-Id: Ia8d5b29418557c947cb603a2070bc927152ee415
Signed-off-by: Daniel Goehring <dgoehrin@os.amperecomputing.com>
---
 tcl/openocd.ian.cfg        | 48 +++++++++++++++++++
 tcl/openocd.mq_1s_pal.cfg  | 82 ++++++++++++++++++++++++++++++++
 tcl/openocd.mq_1s_prot.cfg | 87 ++++++++++++++++++++++++++++++++++
 tcl/openocd.mq_2s_pal.cfg  | 90 ++++++++++++++++++++++++++++++++++++
 tcl/openocd.mq_2s_prot.cfg | 95 ++++++++++++++++++++++++++++++++++++++
 tcl/openocd.qs_1s_pal.cfg  | 81 ++++++++++++++++++++++++++++++++
 tcl/openocd.qs_1s_prot.cfg | 86 ++++++++++++++++++++++++++++++++++
 tcl/openocd.qs_2s_pal.cfg  | 89 +++++++++++++++++++++++++++++++++++
 tcl/openocd.qs_2s_prot.cfg | 94 +++++++++++++++++++++++++++++++++++++
 9 files changed, 752 insertions(+)
 create mode 100644 tcl/openocd.ian.cfg
 create mode 100644 tcl/openocd.mq_1s_pal.cfg
 create mode 100644 tcl/openocd.mq_1s_prot.cfg
 create mode 100644 tcl/openocd.mq_2s_pal.cfg
 create mode 100644 tcl/openocd.mq_2s_prot.cfg
 create mode 100644 tcl/openocd.qs_1s_pal.cfg
 create mode 100644 tcl/openocd.qs_1s_prot.cfg
 create mode 100644 tcl/openocd.qs_2s_pal.cfg
 create mode 100644 tcl/openocd.qs_2s_prot.cfg

diff --git a/tcl/openocd.ian.cfg b/tcl/openocd.ian.cfg
new file mode 100644
index 000000000..0edb69c49
--- /dev/null
+++ b/tcl/openocd.ian.cfg
@@ -0,0 +1,48 @@
+#
+# Example OpenOCD user configuration for the Ampere Altra ("Quicksilver")
+# Development Platform
+#
+# Copyright (c) 2019-2020, Ampere Computing LLC
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of
+# the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program;
+#
+#
+
+#set DPI_ADDRESS 172.29.148.84
+#set DPI_PORT 5555
+source [find tcl/interface/jtag_dpi.cfg]
+reset_config trst_only
+jtag newtap smpro tap -irlen 4 -expected-id 0x4ba00477
+jtag newtap pmpro tap -irlen 4 -expected-id 0x4ba00477
+jtag newtap armv8 tap -irlen 4 -expected-id 0x2ba06477
+dap create smpro.dap -adiv5 -chain-position smpro.tap
+dap create pmpro.dap -adiv5 -chain-position pmpro.tap
+dap create v8.dap -adiv6 -chain-position armv8.tap
+scan_chain
+# Set default AP to APB
+v8.dap apsel 1
+# Create some MEM-AP targets for direct acccess
+target create apb mem_ap -dap v8.dap -ap-num 1
+target create axi mem_ap -dap v8.dap -ap-num 2
+
+# -rtos hwthread allows SMP cores to be viewed as threads (info threads)
+target create smpro cortex_m -dap smpro.dap
+target create pmpro cortex_m -dap pmpro.dap
+cti create cti0 -dap v8.dap -ap-num 1 -baseaddr 0xa2220000
+target create core0 aarch64 -dap v8.dap -ap-num 1 -coreid 0 -cti cti0 -dbgbase 0xa2210000 -rtos hwthread
+#
+#cti create cti1 -dap v8.dap -ap-num 1 -baseaddr 0xa2320000
+#target create core1 aarch64 -dap v8.dap -ap-num 1 -coreid 1 -cti cti1 -dbgbase 0xa2310000 -rtos hwthread
+#target smp core0 core1
+poll off
diff --git a/tcl/openocd.mq_1s_pal.cfg b/tcl/openocd.mq_1s_pal.cfg
new file mode 100644
index 000000000..9a74a10c5
--- /dev/null
+++ b/tcl/openocd.mq_1s_pal.cfg
@@ -0,0 +1,82 @@
+#
+# Example OpenOCD user configuration for the Ampere Altra Max ("Mystique")
+# Development Platform
+#
+# Copyright (c) 2020-2021, Ampere Computing LLC
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of
+# the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program;
+#
+#
+
+# Command Line Argument Description
+#
+# DPI_PORT
+# Set the DPI JTAG server port assigned to the emulator BFM transactor
+# Syntax: -c "set DPI_PORT {5555}"
+#
+# SYSNAME
+# Override the default system name
+# If not specified, defaults to "mq"
+# Syntax: -c "set SYSNAME {mq}"
+#
+# Life-Cycle State (LCS)
+# If not specified, defaults to "Secure LCS"
+# LCS=0, "Secure LCS"
+# LCS=1, "Chip Manufacturing LCS"
+# Syntax: -c "set LCS {0}"
+# Syntax: -c "set LCS {1}"
+#
+# CORELIST_S0
+# Specify available physical cores by number
+# Example syntax to connect to physical cores 16 and 17 for S0
+# Syntax: -c "set CORELIST_S0 {16 17}"
+#
+# COREMASK_S0_LO
+# Specify available physical cores 0-63 by mask
+# Example syntax to connect to physical cores 16 and 17 for S0
+# Syntax: -c "set COREMASK_S0_LO {0x0000000000030000}"
+#
+# COREMASK_S0_HI
+# Specify available physical cores 64 and above by mask
+# Example syntax to connect to physical cores 94 and 95 for S0
+# Syntax: -c "set COREMASK_S0_HI {0x00000000C0000000}"
+#
+# PHYS_IDX
+# Enable OpenOCD ARMv8 core target physical indexing
+# If not specified, defaults to OpenOCD ARMv8 core target logical indexing
+# Syntax: -c "set PHYS_IDX {}"
+
+bindto 0.0.0.0
+#debug_level 3
+
+#gdb_port 3333
+#telnet_port 4444
+#tcl_port 6666
+
+#
+# Interface Configuration
+#
+
+source [find interface/jtag_dpi.cfg]
+
+#
+# Board (Platform) Configuration
+#
+
+if { ![info exists SYSNAME] } {
+	set SYSNAME mq
+}
+
+set MQ_ENABLE ""
+source [find board/ampere_qs_mq_1s.cfg]
diff --git a/tcl/openocd.mq_1s_prot.cfg b/tcl/openocd.mq_1s_prot.cfg
new file mode 100644
index 000000000..1a5155263
--- /dev/null
+++ b/tcl/openocd.mq_1s_prot.cfg
@@ -0,0 +1,87 @@
+#
+# Example OpenOCD user configuration for the Ampere Altra Max ("Mystique")
+# Development Platform
+#
+# Copyright (c) 2020-2021, Ampere Computing LLC
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of
+# the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program;
+#
+#
+
+# Command Line Argument Description
+#
+# JTAGFREQ
+# Override the default JTAG clock frequency
+# If not specified, defaults to "100" kHz
+# Syntax: -c "set JTAGFREQ {100}"
+#
+# SYSNAME
+# Override the default system name
+# If not specified, defaults to "mq"
+# Syntax: -c "set SYSNAME {mq}"
+#
+# Life-Cycle State (LCS)
+# If not specified, defaults to "Secure LCS"
+# LCS=0, "Secure LCS"
+# LCS=1, "Chip Manufacturing LCS"
+# Syntax: -c "set LCS {0}"
+# Syntax: -c "set LCS {1}"
+#
+# CORELIST_S0
+# Specify available physical cores by number
+# Example syntax to connect to physical cores 16 and 17 for S0
+# Syntax: -c "set CORELIST_S0 {16 17}"
+#
+# COREMASK_S0_LO
+# Specify available physical cores 0-63 by mask
+# Example syntax to connect to physical cores 16 and 17 for S0
+# Syntax: -c "set COREMASK_S0_LO {0x0000000000030000}"
+#
+# COREMASK_S0_HI
+# Specify available physical cores 64 and above by mask
+# Example syntax to connect to physical cores 94 and 95 for S0
+# Syntax: -c "set COREMASK_S0_HI {0x00000000C0000000}"
+#
+# PHYS_IDX
+# Enable OpenOCD ARMv8 core target physical indexing
+# If not specified, defaults to OpenOCD ARMv8 core target logical indexing
+# Syntax: -c "set PHYS_IDX {}"
+
+bindto 0.0.0.0
+#debug_level 3
+
+#gdb_port 3333
+#telnet_port 4444
+#tcl_port 6666
+
+#
+# Interface Configuration
+#
+
+source [find interface/ftdi/olimex-arm-usb-ocd-h.cfg]
+
+#
+# Board (Platform) Configuration
+#
+
+if { ![info exists JTAGFREQ] } {
+	set JTAGFREQ 100
+}
+
+if { ![info exists SYSNAME] } {
+	set SYSNAME mq
+}
+
+set MQ_ENABLE ""
+source [find board/ampere_qs_mq_1s.cfg]
diff --git a/tcl/openocd.mq_2s_pal.cfg b/tcl/openocd.mq_2s_pal.cfg
new file mode 100644
index 000000000..a38a565d4
--- /dev/null
+++ b/tcl/openocd.mq_2s_pal.cfg
@@ -0,0 +1,90 @@
+#
+# Example OpenOCD user configuration for the Ampere Altra Max ("Mystique")
+# Development Platform
+#
+# Copyright (c) 2020-2021, Ampere Computing LLC
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of
+# the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program;
+#
+#
+
+# Command Line Argument Description
+#
+# DPI_PORT
+# Set the DPI JTAG server port assigned to the emulator BFM transactor
+# Syntax: -c "set DPI_PORT {5555}"
+#
+# SYSNAME
+# Override the default system name
+# If not specified, defaults to "mq"
+# Syntax: -c "set SYSNAME {mq}"
+#
+# Life-Cycle State (LCS)
+# If not specified, defaults to "Secure LCS"
+# LCS=0, "Secure LCS"
+# LCS=1, "Chip Manufacturing LCS"
+# Syntax: -c "set LCS {0}"
+# Syntax: -c "set LCS {1}"
+#
+# CORELIST_S0, CORELIST_S1
+# Specify available physical cores by number
+# Example syntax to connect to physical cores 16 and 17 for S0 and S1
+# Syntax: -c "set CORELIST_S0 {16 17}"
+# Syntax: -c "set CORELIST_S1 {16 17}"
+#
+# COREMASK_S0_LO, COREMASK_S1_LO
+# Specify available physical cores 0-63 by mask
+# Example syntax to connect to physical cores 16 and 17 for S0 and S1
+# Syntax: -c "set COREMASK_S0_LO {0x0000000000030000}"
+# Syntax: -c "set COREMASK_S1_LO {0x0000000000030000}"
+#
+# COREMASK_S0_HI, COREMASK_S1_HI
+# Specify available physical cores 64 and above by mask
+# Example syntax to connect to physical cores 94 and 95 for S0 and S1
+# Syntax: -c "set COREMASK_S0_HI {0x00000000C0000000}"
+# Syntax: -c "set COREMASK_S1_HI {0x00000000C0000000}"
+#
+# SPLITSMP
+# Group all ARMv8 cores per socket into individual SMP sessions
+# If not specified, group ARMv8 cores from both sockets into one SMP session
+# Syntax: -c "set SPLITSMP {}"
+#
+# PHYS_IDX
+# Enable OpenOCD ARMv8 core target physical indexing
+# If not specified, defaults to OpenOCD ARMv8 core target logical indexing
+# Syntax: -c "set PHYS_IDX {}"
+
+bindto 0.0.0.0
+#debug_level 3
+
+#gdb_port 3333
+#telnet_port 4444
+#tcl_port 6666
+
+#
+# Interface Configuration
+#
+
+source [find interface/jtag_dpi.cfg]
+
+#
+# Board (Platform) Configuration
+#
+
+if { ![info exists SYSNAME] } {
+	set SYSNAME mq
+}
+
+set MQ_ENABLE ""
+source [find board/ampere_qs_mq_2s.cfg]
diff --git a/tcl/openocd.mq_2s_prot.cfg b/tcl/openocd.mq_2s_prot.cfg
new file mode 100644
index 000000000..89f7e0ce9
--- /dev/null
+++ b/tcl/openocd.mq_2s_prot.cfg
@@ -0,0 +1,95 @@
+#
+# Example OpenOCD user configuration for the Ampere Altra Max ("Mystique")
+# Development Platform
+#
+# Copyright (c) 2020-2021, Ampere Computing LLC
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of
+# the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program;
+#
+#
+
+# Command Line Argument Description
+#
+# JTAGFREQ
+# Override the default JTAG clock frequency
+# If not specified, defaults to "100" kHz
+# Syntax: -c "set JTAGFREQ {100}"
+#
+# SYSNAME
+# Override the default system name
+# If not specified, defaults to "mq"
+# Syntax: -c "set SYSNAME {mq}"
+#
+# Life-Cycle State (LCS)
+# If not specified, defaults to "Secure LCS"
+# LCS=0, "Secure LCS"
+# LCS=1, "Chip Manufacturing LCS"
+# Syntax: -c "set LCS {0}"
+# Syntax: -c "set LCS {1}"
+#
+# CORELIST_S0, CORELIST_S1
+# Specify available physical cores by number
+# Example syntax to connect to physical cores 16 and 17 for S0 and S1
+# Syntax: -c "set CORELIST_S0 {16 17}"
+# Syntax: -c "set CORELIST_S1 {16 17}"
+#
+# COREMASK_S0_LO, COREMASK_S1_LO
+# Specify available physical cores 0-63 by mask
+# Example syntax to connect to physical cores 16 and 17 for S0 and S1
+# Syntax: -c "set COREMASK_S0_LO {0x0000000000030000}"
+# Syntax: -c "set COREMASK_S1_LO {0x0000000000030000}"
+#
+# COREMASK_S0_HI, COREMASK_S1_HI
+# Specify available physical cores 64 and above by mask
+# Example syntax to connect to physical cores 94 and 95 for S0 and S1
+# Syntax: -c "set COREMASK_S0_HI {0x00000000C0000000}"
+# Syntax: -c "set COREMASK_S1_HI {0x00000000C0000000}"
+#
+# SPLITSMP
+# Group all ARMv8 cores per socket into individual SMP sessions
+# If not specified, group ARMv8 cores from both sockets into one SMP session
+# Syntax: -c "set SPLITSMP {}"
+#
+# PHYS_IDX
+# Enable OpenOCD ARMv8 core target physical indexing
+# If not specified, defaults to OpenOCD ARMv8 core target logical indexing
+# Syntax: -c "set PHYS_IDX {}"
+
+bindto 0.0.0.0
+#debug_level 3
+
+#gdb_port 3333
+#telnet_port 4444
+#tcl_port 6666
+
+#
+# Interface Configuration
+#
+
+source [find interface/ftdi/olimex-arm-usb-ocd-h.cfg]
+
+#
+# Board (Platform) Configuration
+#
+
+if { ![info exists JTAGFREQ] } {
+	set JTAGFREQ 100
+}
+
+if { ![info exists SYSNAME] } {
+	set SYSNAME mq
+}
+
+set MQ_ENABLE ""
+source [find board/ampere_qs_mq_2s.cfg]
diff --git a/tcl/openocd.qs_1s_pal.cfg b/tcl/openocd.qs_1s_pal.cfg
new file mode 100644
index 000000000..6d556fda8
--- /dev/null
+++ b/tcl/openocd.qs_1s_pal.cfg
@@ -0,0 +1,81 @@
+#
+# Example OpenOCD user configuration for the Ampere Altra ("Quicksilver")
+# Development Platform
+#
+# Copyright (c) 2019-2021, Ampere Computing LLC
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of
+# the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program;
+#
+#
+
+# Command Line Argument Description
+#
+# DPI_PORT
+# Set the DPI JTAG server port assigned to the emulator BFM transactor
+# Syntax: -c "set DPI_PORT {5555}"
+#
+# SYSNAME
+# Override the default system name
+# If not specified, defaults to "qs"
+# Syntax: -c "set SYSNAME {qs}"
+#
+# Life-Cycle State (LCS)
+# If not specified, defaults to "Secure LCS"
+# LCS=0, "Secure LCS"
+# LCS=1, "Chip Manufacturing LCS"
+# Syntax: -c "set LCS {0}"
+# Syntax: -c "set LCS {1}"
+#
+# CORELIST_S0
+# Specify available physical cores by number
+# Example syntax to connect to physical cores 16 and 17 for S0
+# Syntax: -c "set CORELIST_S0 {16 17}"
+#
+# COREMASK_S0_LO
+# Specify available physical cores 0-63 by mask
+# Example syntax to connect to physical cores 16 and 17 for S0
+# Syntax: -c "set COREMASK_S0_LO {0x0000000000030000}"
+#
+# COREMASK_S0_HI
+# Specify available physical cores 64 and above by mask
+# Example syntax to connect to physical cores 78 and 79 for S0
+# Syntax: -c "set COREMASK_S0_HI {0x000000000000C000}"
+#
+# PHYS_IDX
+# Enable OpenOCD ARMv8 core target physical indexing
+# If not specified, defaults to OpenOCD ARMv8 core target logical indexing
+# Syntax: -c "set PHYS_IDX {}"
+
+bindto 0.0.0.0
+#debug_level 3
+
+#gdb_port 3333
+#telnet_port 4444
+#tcl_port 6666
+
+#
+# Interface Configuration
+#
+
+source [find interface/jtag_dpi.cfg]
+
+#
+# Board (Platform) Configuration
+#
+
+if { ![info exists SYSNAME] } {
+	set SYSNAME qs
+}
+
+source [find board/ampere_qs_mq_1s.cfg]
diff --git a/tcl/openocd.qs_1s_prot.cfg b/tcl/openocd.qs_1s_prot.cfg
new file mode 100644
index 000000000..bc262052d
--- /dev/null
+++ b/tcl/openocd.qs_1s_prot.cfg
@@ -0,0 +1,86 @@
+#
+# Example OpenOCD user configuration for the Ampere Altra ("Quicksilver")
+# Development Platform
+#
+# Copyright (c) 2019-2021, Ampere Computing LLC
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of
+# the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program;
+#
+#
+
+# Command Line Argument Description
+#
+# JTAGFREQ
+# Override the default JTAG clock frequency
+# If not specified, defaults to "100" kHz
+# Syntax: -c "set JTAGFREQ {100}"
+#
+# SYSNAME
+# Override the default system name
+# If not specified, defaults to "qs"
+# Syntax: -c "set SYSNAME {qs}"
+#
+# Life-Cycle State (LCS)
+# If not specified, defaults to "Secure LCS"
+# LCS=0, "Secure LCS"
+# LCS=1, "Chip Manufacturing LCS"
+# Syntax: -c "set LCS {0}"
+# Syntax: -c "set LCS {1}"
+#
+# CORELIST_S0
+# Specify available physical cores by number
+# Example syntax to connect to physical cores 16 and 17 for S0
+# Syntax: -c "set CORELIST_S0 {16 17}"
+#
+# COREMASK_S0_LO
+# Specify available physical cores 0-63 by mask
+# Example syntax to connect to physical cores 16 and 17 for S0
+# Syntax: -c "set COREMASK_S0_LO {0x0000000000030000}"
+#
+# COREMASK_S0_HI
+# Specify available physical cores 64 and above by mask
+# Example syntax to connect to physical cores 78 and 79 for S0
+# Syntax: -c "set COREMASK_S0_HI {0x000000000000C000}"
+#
+# PHYS_IDX
+# Enable OpenOCD ARMv8 core target physical indexing
+# If not specified, defaults to OpenOCD ARMv8 core target logical indexing
+# Syntax: -c "set PHYS_IDX {}"
+
+bindto 0.0.0.0
+#debug_level 3
+
+#gdb_port 3333
+#telnet_port 4444
+#tcl_port 6666
+
+#
+# Interface Configuration
+#
+
+source [find interface/ftdi/olimex-arm-usb-ocd-h.cfg]
+
+#
+# Board (Platform) Configuration
+#
+
+if { ![info exists JTAGFREQ] } {
+	set JTAGFREQ 100
+}
+
+if { ![info exists SYSNAME] } {
+	set SYSNAME qs
+}
+
+source [find board/ampere_qs_mq_1s.cfg]
diff --git a/tcl/openocd.qs_2s_pal.cfg b/tcl/openocd.qs_2s_pal.cfg
new file mode 100644
index 000000000..76d5a79bc
--- /dev/null
+++ b/tcl/openocd.qs_2s_pal.cfg
@@ -0,0 +1,89 @@
+#
+# Example OpenOCD user configuration for the Ampere Altra ("Quicksilver")
+# Development Platform
+#
+# Copyright (c) 2019-2021, Ampere Computing LLC
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of
+# the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program;
+#
+#
+
+# Command Line Argument Description
+#
+# DPI_PORT
+# Set the DPI JTAG server port assigned to the emulator BFM transactor
+# Syntax: -c "set DPI_PORT {5555}"
+#
+# SYSNAME
+# Override the default system name
+# If not specified, defaults to "qs"
+# Syntax: -c "set SYSNAME {qs}"
+#
+# Life-Cycle State (LCS)
+# If not specified, defaults to "Secure LCS"
+# LCS=0, "Secure LCS"
+# LCS=1, "Chip Manufacturing LCS"
+# Syntax: -c "set LCS {0}"
+# Syntax: -c "set LCS {1}"
+#
+# CORELIST_S0, CORELIST_S1
+# Specify available physical cores by number
+# Example syntax to connect to physical cores 16 and 17 for S0 and S1
+# Syntax: -c "set CORELIST_S0 {16 17}"
+# Syntax: -c "set CORELIST_S1 {16 17}"
+#
+# COREMASK_S0_LO, COREMASK_S1_LO
+# Specify available physical cores 0-63 by mask
+# Example syntax to connect to physical cores 16 and 17 for S0 and S1
+# Syntax: -c "set COREMASK_S0_LO {0x0000000000030000}"
+# Syntax: -c "set COREMASK_S1_LO {0x0000000000030000}"
+#
+# COREMASK_S0_HI, COREMASK_S1_HI
+# Specify available physical cores 64 and above by mask
+# Example syntax to connect to physical cores 78 and 79 for S0 and S1
+# Syntax: -c "set COREMASK_S0_HI {0x000000000000C000}"
+# Syntax: -c "set COREMASK_S1_HI {0x000000000000C000}"
+#
+# SPLITSMP
+# Group all ARMv8 cores per socket into individual SMP sessions
+# If not specified, group ARMv8 cores from both sockets into one SMP session
+# Syntax: -c "set SPLITSMP {}"
+#
+# PHYS_IDX
+# Enable OpenOCD ARMv8 core target physical indexing
+# If not specified, defaults to OpenOCD ARMv8 core target logical indexing
+# Syntax: -c "set PHYS_IDX {}"
+
+bindto 0.0.0.0
+#debug_level 3
+
+#gdb_port 3333
+#telnet_port 4444
+#tcl_port 6666
+
+#
+# Interface Configuration
+#
+
+source [find interface/jtag_dpi.cfg]
+
+#
+# Board (Platform) Configuration
+#
+
+if { ![info exists SYSNAME] } {
+	set SYSNAME qs
+}
+
+source [find board/ampere_qs_mq_2s.cfg]
diff --git a/tcl/openocd.qs_2s_prot.cfg b/tcl/openocd.qs_2s_prot.cfg
new file mode 100644
index 000000000..1d812e2bd
--- /dev/null
+++ b/tcl/openocd.qs_2s_prot.cfg
@@ -0,0 +1,94 @@
+#
+# Example OpenOCD user configuration for the Ampere Altra ("Quicksilver")
+# Development Platform
+#
+# Copyright (c) 2019-2021, Ampere Computing LLC
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of
+# the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program;
+#
+#
+
+# Command Line Argument Description
+#
+# JTAGFREQ
+# Override the default JTAG clock frequency
+# If not specified, defaults to "100" kHz
+# Syntax: -c "set JTAGFREQ {100}"
+#
+# SYSNAME
+# Override the default system name
+# If not specified, defaults to "qs"
+# Syntax: -c "set SYSNAME {qs}"
+#
+# Life-Cycle State (LCS)
+# If not specified, defaults to "Secure LCS"
+# LCS=0, "Secure LCS"
+# LCS=1, "Chip Manufacturing LCS"
+# Syntax: -c "set LCS {0}"
+# Syntax: -c "set LCS {1}"
+#
+# CORELIST_S0, CORELIST_S1
+# Specify available physical cores by number
+# Example syntax to connect to physical cores 16 and 17 for S0 and S1
+# Syntax: -c "set CORELIST_S0 {16 17}"
+# Syntax: -c "set CORELIST_S1 {16 17}"
+#
+# COREMASK_S0_LO, COREMASK_S1_LO
+# Specify available physical cores 0-63 by mask
+# Example syntax to connect to physical cores 16 and 17 for S0 and S1
+# Syntax: -c "set COREMASK_S0_LO {0x0000000000030000}"
+# Syntax: -c "set COREMASK_S1_LO {0x0000000000030000}"
+#
+# COREMASK_S0_HI, COREMASK_S1_HI
+# Specify available physical cores 64 and above by mask
+# Example syntax to connect to physical cores 78 and 79 for S0 and S1
+# Syntax: -c "set COREMASK_S0_HI {0x000000000000C000}"
+# Syntax: -c "set COREMASK_S1_HI {0x000000000000C000}"
+#
+# SPLITSMP
+# Group all ARMv8 cores per socket into individual SMP sessions
+# If not specified, group ARMv8 cores from both sockets into one SMP session
+# Syntax: -c "set SPLITSMP {}"
+#
+# PHYS_IDX
+# Enable OpenOCD ARMv8 core target physical indexing
+# If not specified, defaults to OpenOCD ARMv8 core target logical indexing
+# Syntax: -c "set PHYS_IDX {}"
+
+bindto 0.0.0.0
+#debug_level 3
+
+#gdb_port 3333
+#telnet_port 4444
+#tcl_port 6666
+
+#
+# Interface Configuration
+#
+
+source [find interface/ftdi/olimex-arm-usb-ocd-h.cfg]
+
+#
+# Board (Platform) Configuration
+#
+
+if { ![info exists JTAGFREQ] } {
+	set JTAGFREQ 100
+}
+
+if { ![info exists SYSNAME] } {
+	set SYSNAME qs
+}
+
+source [find board/ampere_qs_mq_2s.cfg]
-- 
2.25.1

